### ISSUE TITLE
Interest in simple log adapter?

### ISSUE THREAD

**Issue by killercup** [lucab/libsystemd-rs#37]
I might be interested in adding a simple logger adapter based on the `log` crate using the journald interface here. Is this something that would fit in this crate (behind a feature flag)?

**Comment by lucab** [comment]
Thanks for reaching out. I'm not against that in principle, but at the same time I'm not completely sure this is the right place for that.

The current `logging` module is meant to only provide the low-level "send to journal" primitive (see [review comment](https://github.com/lucab/libsystemd-rs/pull/34#discussion_r347961529)), but we can re-discuss that.

Is it part of a larger logging framework? If so, I do like `slog` and `fern` approach to have the logger  implementation on their side.

**Comment by jabedude** [comment]
Maybe we could have a separate module for implementing the `log` API with journald?

**Comment by killercup** [comment]
Yeah, my idea was to have a have a log module that is opt-in with a feature flag that implements [`log::Log`](https://docs.rs/log/0.4.8/log/trait.Log.html) in a basic way, but maybe with support for journald's native key-value data. If it turns outn to be easy I could also imagine supporting `env_logger::filter::Filter` but in that case you should probably already use a larger logging framework.

**Comment by lucab** [comment]
Ack, then I guess we can add that directly in `logging`, the additional logic seems small enough. We can also just expose it (without a feature flag), as I think we won't incur a lot of additional dependencies.

**Comment by lunaryorn** [comment]
I just published <https://crates.io/crates/systemd-journal-logger> which implements a logger over libsystemd's journal functions.
